eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -603,6 +624,11 @@ public void registerHealthCheck(HealthCheckHandler healthCheckHandler) {
         }
     }
 
+    @Override
+    public void registerEventListener(EurekaEventListener eventListener) {
```

Comment:
On a second thought. Since the new event model is not backward compatible, we should stay with the old one, and depend on constructor injection only. We can introduce the new event model with new EurekaClient implementation, and make it parallel to the old one.